llama_index/vector_stores/google/generativeai/base.py

```diff
@@ -0,0 +1,383 @@
+from llama_index.indices.service_context import ServiceContext
+from llama_index.schema import BaseNode, RelatedNodeInfo, TextNode
+from llama_index.vector_stores.types import (
+    BasePydanticVectorStore,
+    MetadataFilters,
+    VectorStoreQuery,
+    VectorStoreQueryResult,
+)
+
+_logger = logging.getLogger(__name__)
+_import_err_msg = "`google.generativeai` package not found, please run `pip install google-generativeai`"
+_default_doc_id = "default-doc"
+
+
+google_service_context = ServiceContext.from_defaults(
+    # Avoids instantiating OpenAI as the default model.
+    llm=None,
+    # Avoids instantiating HuggingFace as the default model.
+    embed_model=None,
+)
+"""Google GenerativeAI service context.
+
+Use this to provide the correct service context for `GoogleVectorStore`.
+
+See the docstring for `GoogleVectorStore` for usage example.
+"""
+
+
+class GoogleVectorStore(BasePydanticVectorStore):
+    """Google GenerativeAI Vector Store.
+
+    Currently, it computes the embedding vectors on the server side.
+    However, in the near future, a series of embedding models will be
+    released for various applications such as medical, legal, etc.
+
+    Example:
+        google_vector_store = GoogleVectorStore.from_corpus(
+                corpus_id="my-corpus-id")
+        index = VectorStoreIndex.from_vector_store(
+                google_vector_store,
+                service_context=google_service_context)
+
+    Attributes:
+        corpus_id: The corpus ID that this vector store instance will read and
+            write to.
+    """
+
+    # Semantic Retriever stores the document node's text as string and embeds
+    # the vectors on the server automatically.
+    stores_text: bool = True
+    is_embedding_query: bool = False
+
+    # This is not the Google's corpus name but an ID generated in the LlamaIndex
+    # world.
+    corpus_id: str = Field(frozen=True)
+    """Corpus ID that this instance of the vector store is using."""
+
+    _client: Any = PrivateAttr()
+
+    def __init__(self, *, client: Any, **kwargs: Any):
+        """Raw constructor.
+
+        Use the class method `from_corpus` or `create_corpus` instead.
+
+        Args:
+            client: The low-level retriever class from google.ai.generativelanguage.
+        """
+        try:
+            import google.ai.generativelanguage as genai
+        except ImportError:
+            raise ImportError(_import_err_msg)
+
+        super().__init__(**kwargs)
+
+        assert isinstance(client, genai.RetrieverServiceClient)
+        self._client = client
+
+    @classmethod
+    def from_corpus(cls, *, corpus_id: str) -> "GoogleVectorStore":
+        """Create an instance that points to an existing corpus.
+
+        Args:
+            corpus_id: ID of an existing corpus on Google's server.
+
+        Returns:
+            An instance of the vector store that points to the specified corpus.
```

Comment:
Do we return an error if the corpus does not exist / permission denied?

Reply:
Not here. But future operation would produce the failure. I thought of checking for the corpus, but speaking with Michael we want to reduce unnecessary RPC to our backend. So, it's not checked. Let me know what you think.

Reply:
Ok. Added the code to try to fetch the corpus information. Would raise a custom exception if failed.